raster/src/main/scala/geotrellis/raster/io/geotiff/GeoTiffMultibandTile.scala

```diff
@@ -325,31 +325,63 @@ abstract class GeoTiffMultibandTile(
     crop(this.gridBounds)
 
   /**
-   * Performs a crop on itself. The returned MultibandGeoTiffTile will
-   * contain bands that have the same area as the inputte GridBounds.
+   * Crop this tile to given pixel region.
    *
-   * @param  gridBounds  A [[GridBounds]] that contains the area to be cropped.
-   * @return             A [[ArrayMultibandTile]]
+   * @param bounds Pixel bounds specifying the crop area.
    */
-  def crop(gridBounds: GridBounds): ArrayMultibandTile = {
-    val bands = Array.fill(bandCount)(ArrayTile.empty(cellType, gridBounds.width, gridBounds.height))
-    val intersectingSegments = segmentLayout.intersectingSegments(gridBounds)
-    if (hasPixelInterleave) {
-      logger.debug(s"Cropping $gridBounds uses ${intersectingSegments.length} out of ${segmentCount} pixel-interleave segments")
+ def crop(gridBounds: GridBounds): ArrayMultibandTile =
+  crop(List(gridBounds)).next._2
 
-      // de-interlace the pixels from each segment
-      for ((segmentId, segment) <- getSegments(intersectingSegments)) {
-        val segmentBounds = segmentLayout.getGridBounds(segmentId)
-        val segmentTransform = segmentLayout.getSegmentTransform(segmentId, bandCount)
-        val overlap = gridBounds.intersection(segmentBounds).get
+  /**
+   * Crop this tile to given pixel regions.
+   *
+   * @param windows Pixel bounds specifying the crop areas
+   */
+  def crop(windows: Seq[GridBounds]): Iterator[(GridBounds, ArrayMultibandTile)] = {
+    case class Chip(
```

Comment:
I'm confused that it lives inside a function, mb to put into a package object and to make it private?

Reply:
The singleband case has a slightly different `case class Chip`, it really has no meaning outside of this function. Do you think there is a downside to having it here ?

Reply:
Nope, just wanted to be sure that it's here because useless anywhere else